Watchdog changes

wferi opened the pull request:
The first two commits should be okay for needle.
Switching the default maybe not (although I only tested it there).

jfriesse commented:
@wferi First two patches are nice so ACK and I've merged them as 4738bcce37fe495843345e6d60eddb394345c06c / b510a0f45c3a55499e58cc47549a74fe80ff1b39 and 1f3d75464fd6afe34cff0de39e6ff70bf3c92a25 / 151ed9dfe518c89ddf66dc59b0cb15bd30f8b98b.

Last patch is also good but I would rather keep compatibility with configurations which already uses "off". Can you please change the last patch so no key == empty string == off?

wferi commented:
Thanks. I replaced the third patch (and dropped the first two).
By the way, this code feels like `setup_watchdog()` could be called multiple times. If that's really the case, then a `free (watchdog_device)` is also needed right before resetting it.

jfriesse commented:
@wferi Again nice patch so ACK and merged as b7b318b86fc0719b12ec62784692ebaacfc5310e and 3323942702ca6e688270873188a01011452505e4 . This time only for master because needle behavior shouldn't change.

Thank you for the patches.

wferi commented:
Great, thanks! The second change is also valid for needle, though the patch itself won't apply cleanly due to the context change introduced by the first. I can open a separate PR for that if you prefer.

jfriesse commented:
@wferi True. It would be great if you could backport second patch.

wferi commented:
I cherry-picked it into #244.
Shall I go on enabling watchdog support by default if `linux/watchdog.h` is present?

jfriesse commented:
@wferi yep. Idea is to completely remove the configure option, and remove it also from corosync.spec.in. Ans yes, `linux/watchdog.h` is probably best one to check.
